.github/workflows/ci.yaml

```diff
@@ -0,0 +1,53 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - "main"
+  workflow_dispatch:
+
+jobs:
+  test:
+    name: Test on ubuntu-latest, Python ${{ matrix.python-version }}
+    runs-on: "ubuntu-latest"
+    strategy:
+      matrix:
+        python-version: ["3.10", "3.11", "3.12"]
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Additional info about the build
+        shell: bash
+        run: |
+          uname -a
+          df -h
+          ulimit -a
+
+      - name: Setup Python ${{matrix.python-version}}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install Dependencies
+        run: |
+          python -m pip install --upgrade pip setuptools wheel
+          python -m pip install --no-cache-dir .
+          python -m pip install --no-cache-dir '.[test]'
+          cd tests/
```

Comment:
Probably move this down to be part of the `Run tests` step. Also, this will break the current `pytest` command (since it would now be running tests in the `tests/tests/` directory).